src/openssl/digest.c

```diff
@@ -27,13 +27,13 @@
 #include "../digest.h"
 #include "../logger.h"
 
-static digest_t *digest_open(const EVP_MD *evp_md, int maclength) {
+static digest_t *digest_open(const EVP_MD *evp_md, ssize_t maclength) {
```

Comment:
Changing `maclength` to `ssize_t` doesn't really improve things, it still needs a lot of casting. Perhaps it is better to make it `size_t`. While we check for it to be negative, nowhere do we actually set it to a negative value, so I don't think we need to handle negative values at all.

Reply:
`digest_open` is called by `digest_open_by_name` and `digest_open_by_nid`, both of which are called with `maclength == -1`.

https://github.com/gsliepen/tinc/blob/1.1/src/openssl/prf.c#L33

https://github.com/gsliepen/tinc/blob/1.1/src/protocol_auth.c#L642

https://github.com/gsliepen/tinc/blob/1.1/src/protocol_auth.c#L527

Maybe it's best to replace `maclength` with `size_t` and change all these calls to `0`?

Reply:
Ah, I missed that. -1 means using the same size as the digest algorithm produces, 0 means not using a digest. A large value should work as well, like `(size_t)-1`. Maybe a `#define` or `static const` variable could be made for this.